sdk/iot/core/src/az_iot_hub_client.c

```diff
@@ -107,3 +108,40 @@ AZ_NODISCARD az_result az_iot_hub_client_id_get(
 
   return AZ_OK;
 }
+
+AZ_NODISCARD az_result
+az_iot_hub_client_properties_init(az_iot_hub_client_properties* properties, az_span buffer)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_VALID_SPAN(buffer, 0, false);
+
+  properties->_internal.properties = buffer;
+  properties->_internal.current_property = az_span_ptr(buffer);
+
+  return AZ_OK;
+}
+
+AZ_NODISCARD az_result az_iot_hub_client_properties_append(
+    az_iot_hub_client_properties* properties,
+    az_span name,
+    az_span value)
+{
+  AZ_PRECONDITION_NOT_NULL(properties);
+  AZ_PRECONDITION_VALID_SPAN(name, 1, false);
+  AZ_PRECONDITION_VALID_SPAN(value, 1, false);
+
+  az_span prop_span = properties->_internal.properties;
```

Comment:
>az_span prop_span = properties->_internal.properties [](start = 2, length = 52)

is it possible to use pointer?

Reply:
If I make it a pointer I can't have the dynamic we had talked about where you only set the "output" span once everything is successful. In this way, if you try to append a key and value and it would exceed the span, previously appended k/v won't get ruined. The unit tests validate that dynamic [here](https://github.com/Azure/azure-sdk-for-c/blob/f1f7b859b19718500d29cccc96023b5188d4c193/sdk/iot/core/tests/cmocka/az_iot_hub_client_tests.c#L377-L384).